test/php/Nominatim/ParameterParserTest.php

```diff
@@ -0,0 +1,216 @@
+
+
+    public function testGetInt()
+    {
+        $oParams = new ParameterParser([
+                                        'int1' => '5',
+                                        'int2' => '-1',
+                                        'int3' => '',
+                                        'int4' => 0
+                                       ]);
+
+        $this->assertSame(false, $oParams->getInt('non-exists'));
+        $this->assertSame(999, $oParams->getInt('non-exists', 999));
+        $this->assertSame(5, $oParams->getInt('int1'));
+
+        $this->assertSame(-1, $oParams->getInt('int2'));
+        $this->assertSame(false, $oParams->getInt('int3')); // FIXME: should be 0 instead?
+        $this->assertSame(0, $oParams->getInt('int4'));
+    }
+
+
+    public function testGetIntWithEmpytString()
+    {
+        $this->setExpectedException(Exception::class, "Integer number expected for parameter 'int5'");
+        (new ParameterParser(['int5' => 'a']))->getInt('int5');
+    }
+
+
+    public function testGetFloat()
+    {
+
+        $oParams = new ParameterParser([
+                                        'float1' => '1.0',
+                                        'float2' => '-5',
+                                        'float3' => '',
+                                        'float4' => 0
+                                       ]);
+
+        $this->assertSame(false, $oParams->getFloat('non-exists'));
+        $this->assertSame(999, $oParams->getFloat('non-exists', 999));
+        $this->assertSame(1.0, $oParams->getFloat('float1'));
+        $this->assertSame(-5.0, $oParams->getFloat('float2'));
+        $this->assertSame(false, $oParams->getFloat('float3')); // FIXME: should be 0 instead?
+        $this->assertSame(0.0, $oParams->getFloat('float4'));
+    }
+
+
+    public function testGetFloatWithString()
+    {
+        $this->setExpectedException(Exception::class, "Floating-point number expected for parameter 'float5'");
+        (new ParameterParser(['float5' => 'a']))->getFloat('float5');
+    }
+
+
+    public function testGetFloatWithInvalidNumber()
+    {
+        $this->setExpectedException(Exception::class, "Floating-point number expected for parameter 'float6'");
+        (new ParameterParser(['float6' => '-55.']))->getFloat('float6');
+    }
+
+
+    public function testGetString()
+    {
+        $oParams = new ParameterParser([
+                                        'str1' => 'abc',
+                                        'str2' => '',
+                                        'str3' => '0'
+                                       ]);
+
+        $this->assertSame(false, $oParams->getString('non-exists'));
+        $this->assertSame('default', $oParams->getString('non-exists', 'default'));
+        $this->assertSame('abc', $oParams->getString('str1'));
+        $this->assertSame(false, $oParams->getStringList('str2'));
+        $this->assertSame(false, $oParams->getStringList('str3')); // FIXME: should be 0 instead?
```

Comment:
yes

Reply:
PHP does some magic where even returning a '0' gets treated as false during the comparison. I added a note. Starting PHP7 we can define the return value of functions.